runtime/onert/core/include/util/OnertException.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_UTIL_ONERTEXCEPTION_H__
+#define __ONERT_UTIL_ONERTEXCEPTION_H__
+
+#include <iostream>
+
+namespace onert
+{
+namespace util
+{
+
+class OnertException : public std::exception
+{
+  std::string message;
+
+public:
+  OnertException(std::string _m) : message(_m) {}
+
+  virtual const char *what() const throw() { return message.c_str(); }
+};
+
+class ExecutionFailedException : public OnertException
```

Comment:
It seems like this is a super exception for all, e.g., when an op is not yet implemented, 
`ExecutionFailedException` can be thrown or also `NotYetImplementedException` can be thrown.
So we may need some guideline when to use `ExecutionFailedException` rather than `NotYetImplementedException`.

Reply:
Rethinking.. ExecutionFailed seems to contain all other exceptions.

So I think you can delete ExecutionFailed.

Reply:
Some questions I have are...

I wonder what exception i have to use while writing code:
1) when system fails to alloc memory
2) whey system fails to find a tensor in map
...

I thought that `ExecutionFailedException` fits for such cases. 

However, when an input value or shape is wrong, exception will fail. Does that mean `InvalidValueException` should inherit `ExecutionFailedException`?

FYI, due to exception design needs consideration of complex cases, _one_ compiler team decided to use only two types of exceptions. :-P UserError (named `OopsException` that needs to be reported to users) and InternalError (internal logic error). :-)

Reply:
I think we need different kind of exceptions. As runtime is getting complicated it is not trivial to know which exception is thrown at which point. And we are going to handle the errors differently.

For example, 

Sometimes we need to report the error to the user.

https://github.com/Samsung/ONE/blob/6735e767658b604e91876160641537dc84a87575/runtime/onert/core/src/exec/ExecutorBase.cc#L263-L264

Then we could throw `InsufficientOutputSizeError` rather than `runtime_error` and it may be caught in `catch (InsufficientOutputSizeError e) { return NNFW_INSUFFICIENT_OUTPUT_SIZE; }` of `nnfw_run`. And in another part of execution it could throw something else.

% When I left a comment "ExecutionFailedException", I did not had many thoughts and it looks too ambiguous 😓

Reply:
I agree with the opinion that ExecutionFailedException is too ambiguous.
I think it is important to identify the detailed cause when executing ExecutionFail.

Therefore, ExecutionFailedException was removed and more detailed Exceptions were added.